presto-benchmark-runner/src/main/java/com/facebook/presto/benchmark/event/BenchmarkQueryEvent.java

```diff
@@ -0,0 +1,108 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.benchmark.event;
+
+import com.facebook.airlift.event.client.EventField;
+import com.google.common.collect.ImmutableList;
+
+import java.util.List;
+import java.util.Optional;
+
+import static java.util.Objects.requireNonNull;
+
+public class BenchmarkQueryEvent
+{
+    public enum EventStatus
+    {
+        SUCCEEDED,
+        FAILED,
+    }
+
+    private final String testId;
+    private final String name;
+    private final String status;
+    private final BenchmarkQueryInfo queryInfo;
+    private final String errorCode;
+    private final String errorMessage;
+    private final BenchmarkQueryFailure finalQueryFailure;
+    private final List<BenchmarkQueryFailure> queryFailures;
+
+    public BenchmarkQueryEvent(
+            String testId,
+            String name,
+            EventStatus status,
+            Optional<BenchmarkQueryInfo> queryInfo,
+            Optional<String> errorCode,
+            Optional<String> errorMessage,
+            Optional<BenchmarkQueryFailure> finalQueryFailure,
+            List<BenchmarkQueryFailure> queryFailures)
+    {
+        this.testId = requireNonNull(testId, "testId is null");
+        this.name = requireNonNull(name, "name is null");
+        this.status = status.name();
+        this.queryInfo = queryInfo.orElse(null);
+        this.errorCode = errorCode.orElse(null);
+        this.errorMessage = errorMessage.orElse(null);
+        this.finalQueryFailure = finalQueryFailure.orElse(null);
+        this.queryFailures = ImmutableList.copyOf(queryFailures);
+    }
+
+    @EventField
+    public String getTestId()
+    {
+        return testId;
+    }
+
+    @EventField
+    public String getName()
+    {
+        return name;
+    }
+
+    @EventField
+    public String getStatus()
+    {
+        return status;
+    }
+
+    @EventField
+    public BenchmarkQueryInfo getQueryInfo()
+    {
+        return queryInfo;
+    }
+
+    @EventField
+    public String getErrorCode()
+    {
+        return errorCode;
+    }
+
+    @EventField
+    public String getErrorMessage()
+    {
+        return errorMessage;
+    }
+
+    @EventField
+    public BenchmarkQueryFailure getFinalQueryFailure()
+    {
+        return finalQueryFailure;
+    }
+
+    @EventField
+    public List<BenchmarkQueryFailure> getQueryFailures()
+    {
+        return queryFailures;
+    }
```

Comment:
You need to first try to understand what was the purpose of having those query failures exported in the output before attempting to copy them from the Verifier code.